src/components/dropdown/dropdown.tsx

```diff
@@ -0,0 +1,192 @@
+import { DropdownStyle, OptionGroupStyle, OptionStyle } from './styles';
+import { SizeType } from '../../core/types/size';
+import IonIcon from '../icons';
+import React, { useState } from 'react';
+
+type SelectSizes = Extract<SizeType, 'sm' | 'md'>;
+
+export interface OptionProps {
+  disabled?: boolean;
+  label: string;
+  selected?: boolean;
+  value: string | number;
+}
+
+export interface OptionGroupProps {
+  disabled?: boolean;
+  options: OptionProps[];
+  label: string;
+}
+
+export interface DropdownProps {
+  disabled?: boolean;
+  multiple?: boolean;
+  options?: OptionProps[];
+  optionsGroup?: OptionGroupProps[];
+  selectedOptions?: OptionProps[];
+  size?: SelectSizes;
+  onSelectedOption: Function;
+}
+
+export interface OptionsConfig extends DropdownProps {
+  indexGroup?: number;
+}
+
+const isDisabled = (option: OptionProps, disabled?: boolean): boolean => {
+  if (disabled || option.disabled) return true;
+  return false;
+};
+const isSelectingNotMultiple = (
```

Comment:
Not preffer negative sentence, change name and logic to positive, is more easy to read.
```suggestion
const isSelectingMultiple = (
```

Reply:
I've tryna, but occurred an bug, when I deselect the last option left selected: it just don't unselect. Other bug: when multiple mode and I select multiple options, I try to unselect one of them, but instead of that, it unselect all the other selected, but don't unselect itself. So I'll keep this function as is it.

Reply:
You should change the name and the value to work correctly